File: 3164-last-visited-integers/last-visited-integers.go
func lastVisitedIntegers(nums []int) []int {
    seen := []int{}
    ans := []int{}
    consecutiveNegatives := 0 

    for _, num := range nums {
        if num > 0 {
            seen = append([]int{num}, seen...)
            consecutiveNegatives = 0 
        } else {
            consecutiveNegatives++ 
            if consecutiveNegatives <= len(seen) {
                ans = append(ans, seen[consecutiveNegatives-1])
            } else {
                ans = append(ans, -1)
            }
        }
    }
    return ans
}

File: 3164-last-visited-integers/last-visited-integers.rs
use std::collections::VecDeque;

impl Solution {
    pub fn last_visited_integers(nums: Vec<i32>) -> Vec<i32> {
        let mut seen = VecDeque::new();
        let mut ans = Vec::new();
        let mut consecutive_negatives = 0;

        for &num in nums.iter() {
            if num == -1 {
                consecutive_negatives += 1;
                if consecutive_negatives <= seen.len() {
                    ans.push(*seen.get(consecutive_negatives - 1).unwrap());
                } else {
                    ans.push(-1);
                }
            } else {
                consecutive_negatives = 0;
                seen.push_front(num);
            }
        }

        ans
    }
}
